toksearch/signal/mds.py

```diff
@@ -196,7 +202,28 @@ def cleanup(self):
         MdsTreeRegistry().close_all_trees()
 
 
-class MdsSignal(Signal):
+def MdsSignal(expression: str,
+              treename: str,
+              location: Optional[Union[str, MdsTreePath]] = None,
+              dims: Iterable[str] = ("times",),
+              data_order: Optional[Iterable[str]] = None,
+              fetch_units: bool = True):
+    """
+    Factory method for _MdsSignal class. Automatically returns
+    a PtDataSignal object if the expression redirects the MDSplus
+    signal to ptdata. Otherwise returns a MdsSignal instance
+    """
+    if PtDataSignal is None or ("ptdata" not in expression and "ptheader" not in expresion):
+        return _MdsSignal(expression, treename, location, dims, 
+                          data_order, fetch_units)
+    else:
+        start = expression.find("(")
+        end = expression.rfind("(")
+        ptdata_pointname = expression[start:end]
+        return PtDataSignal(ptdata_pointname, remote=True, 
```

Comment:
We probably need to derive `remote` from `location` but I don't know how to do this.